scp/cc/cpio/client_providers/instance_client_provider/src/azure/azure_instance_client_provider.cc

```diff
@@ -0,0 +1,149 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "azure_instance_client_provider.h"
+
+#include "absl/log/check.h"
+#include "scp/cc/core/interface/errors.h"
+
+using google::cmrt::sdk::instance_service::v1::
+    GetCurrentInstanceResourceNameRequest;
+using google::cmrt::sdk::instance_service::v1::
+    GetCurrentInstanceResourceNameResponse;
+using google::cmrt::sdk::instance_service::v1::
+    GetInstanceDetailsByResourceNameRequest;
+using google::cmrt::sdk::instance_service::v1::
+    GetInstanceDetailsByResourceNameResponse;
+using google::cmrt::sdk::instance_service::v1::GetTagsByResourceNameRequest;
+using google::cmrt::sdk::instance_service::v1::GetTagsByResourceNameResponse;
+using google::cmrt::sdk::instance_service::v1::InstanceDetails;
+using google::cmrt::sdk::instance_service::v1::
+    ListInstanceDetailsByEnvironmentRequest;
+using google::cmrt::sdk::instance_service::v1::
+    ListInstanceDetailsByEnvironmentResponse;
+using google::scp::core::AsyncContext;
+using google::scp::core::AsyncExecutorInterface;
+using google::scp::core::ExecutionResult;
+using google::scp::core::HttpClientInterface;
+using google::scp::core::SuccessExecutionResult;
+using google::scp::core::FailureExecutionResult;
+
+namespace google::scp::cpio::client_providers {
+
+inline constexpr char kOperatorTagName[] = "operator";
+inline constexpr char kEnvironmentTagName[] = "environment";
+inline constexpr char kServiceTagName[] = "service";
+// Dummy values
+inline constexpr char kResourceNameValue[] = "azure_instance_resource_name";
+inline constexpr char kInstanceId[] = "azure_instance_id";
+inline constexpr char kOperatorTagValue[] = "azure_operator";
+inline constexpr char kEnvironmentTagValue[] = "azure_environment";
+inline constexpr char kServiceTagValue[] = "azure_service";
+
+AzureInstanceClientProvider::AzureInstanceClientProvider() {}
+
+ExecutionResult AzureInstanceClientProvider::Init() noexcept {
+  return SuccessExecutionResult();
+}
+
+ExecutionResult AzureInstanceClientProvider::Run() noexcept {
+  return SuccessExecutionResult();
+}
+
+ExecutionResult AzureInstanceClientProvider::Stop() noexcept {
+  return SuccessExecutionResult();
+}
+
+ExecutionResult AzureInstanceClientProvider::GetCurrentInstanceResourceNameSync(
+    std::string& resource_name) noexcept {
+  // Not implemented.
+  return FailureExecutionResult(SC_UNKNOWN);
+}
+
+ExecutionResult AzureInstanceClientProvider::GetCurrentInstanceResourceName(
+    AsyncContext<GetCurrentInstanceResourceNameRequest,
+                 GetCurrentInstanceResourceNameResponse>&
+        get_resource_name_context) noexcept {
+  get_resource_name_context.response =
+        std::make_shared<GetCurrentInstanceResourceNameResponse>();
+  // We need to figure out what we should return here.
+  get_resource_name_context.response->set_instance_resource_name(kResourceNameValue);
```

Comment:
**[Comment for actual PR]**
This class currently returns fake values or `FailureExecutionResult(SC_UNKNOWN)`. We will implement them in a future PR.